### DISCUSSION TITLE
How to get the name or namespace from a UUID in vSAN?

### DISCUSSION THREAD

**Discussion by bsdnet** [vmware/govmomi#3176] [Community Help]
When creating a directory in vSANDatastore, the top directory is converted into a UUID internally. When use vm.info, we see the vmfs://[vsanDatastoreUUID][directoryUUID][...].

Wonder any API to retrieve the directory name or namespace in directoryUUID.

Any document is appreciated!

**Comment by ekaanbj** [comment]
I was here to find similar information. When listing datastore you get both uuid and name of folder. But nothing that matches the two. Or at least not what I have found.
I wrote a bash function to list folders by name. One could match both. And look at contents to do the matching.

# govc datastore variable
govc_datastore(){
  if [ "$#" -ne 1 ] ; then
    echo "Usage: ${FUNCNAME} datastore"
    govc find . -type s
    return 1
  fi
  export GOVC_DATASTORE=${1}
}
# list datastore.
govc_datastore_list(){
  if [ -z "${GOVC_DATASTORE}" ] ; then
    govc_datastore
  fi
  govc datastore.ls . | grep -v '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | sort | while read DATASTORE ; do 
    echo '*' "${DATASTORE}"
    govc datastore.ls "${DATASTORE}"
  done
}
